notebooks/NIRISS_WFSS_advanced/01_niriss_wfss_image2_image3.ipynb

```diff
@@ -0,0 +1,1026 @@
+    "for img2_asn in image2_asns:\n",
+    "    # check if the calibrated file already exists\n",
+    "    asn_data = json.load(open(img2_asn))\n",
+    "    cal_file = f\"{asn_data['products'][0]['name']}_cal.fits\"\n",
+    "    if os.path.exists(cal_file):\n",
+    "        print(cal_file, 'cal file already exists.')\n",
+    "        continue\n",
+    "    # if not, calibrated with image2\n",
+    "    img2 = Image2Pipeline.call(img2_asn, save_results=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fe1ae678",
+   "metadata": {},
+   "source": [
+    "<a id='default_image3'></a>\n",
+    "### Run Default Image3 "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e1fd5c10",
+   "metadata": {},
+   "source": [
+    "#### Looking in a Level 3 Association File\n",
+    "The contents are quite similar to image2, but notice now that there are many more members that are associated together, and they use the `cal.fits` files from image2."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ccf6ac3b-333f-451d-8157-f83ccba9352c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image3_asns = glob.glob('*image3*asn*.json')\n",
+    "print(len(image3_asns), 'Image3 ASN files')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "add4839e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## look at one of the association files\n",
+    "image3_asn_data = json.load(open(image3_asns[0]))\n",
+    "for key, data in image3_asn_data.items():\n",
+    "    print(f\"{key} : {data}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2831ae47-5b1b-4d92-b02a-8ce1ee359710",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## in particular, take a closer look at the product filenames with the association file:\n",
+    "for product in image3_asn_data['products']:\n",
+    "    for key, value in product.items():\n",
+    "        if key == 'members':\n",
+    "            print(f\"{key}:\")\n",
+    "            for member in value:\n",
+    "                print(f\"    {member['expname']} {member['exptype']}\")\n",
+    "        else:\n",
+    "            print(f\"{key}: {value}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0cab171d",
+   "metadata": {},
+   "source": [
+    "#### Run image3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ca3bd322-7f67-43c4-8dae-dc47779462ee",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "Image3 is where we can make some adjustments to obtain a better output source catalog. The `cal.fits` files will be calibrated into a single combined `i2d.fits` image. More information about the steps performed in the Image3 part of the pipeline can be found in the [Image3 pipeline documentation](https://jwst-pipeline.readthedocs.io/en/latest/jwst/pipeline/calwebb_image3.html).\n",
```

Comment:
Please be a little more specific about "make some adjustments to obtain a better output source catalog".

Reply:
@camipacifici it would actually be really helpful if you could add/if you let me know I can add why you chose the parameters you did for source_catalog and tweakreg. I didn't change what you had in there before, and they look better, but I don't actually know all of the secrets. I did update the text above that call to point to some documentation about possible parameters to change (this accidentally was cut and jupyter overwrote the save before I could bring it back).